RecoMuon/TrackingTools/src/MuonServiceProxy.cc

```diff
@@ -112,23 +104,35 @@ void MuonServiceProxy::update(const edm::EventSetup& setup) {
   if (newCacheId_GTG != theCacheId_GTG) {
     LogTrace(metname) << "GlobalTrackingGeometry changed!";
     theCacheId_GTG = newCacheId_GTG;
-    setup.get<GlobalTrackingGeometryRecord>().get(theTrackingGeometry);
+    if (globalTrackingGeometryToken_.isInitialized()) {
+      theTrackingGeometry = setup.getHandle(globalTrackingGeometryToken_);
```

Comment:
I guess there better be some "FIXME" here to clarify that this branch is needed only as long as the deprecated constructor exists (this applies to other cases below)

Reply:
Done. I added the FIXME comments.

I'd like to get this PR merged first and see that all is OK and that L1s approve of the new version. But after thinking about it more, I realize that I should probably just go ahead fix all the modules using MuonServiceProxy. Most likely many of those modules are not used and therefore will never go through the esConsumes migration and then this deprecated constructor remains forever because who else is ever going to fix it...